rows.go

```diff
@@ -3,19 +3,32 @@ package sqlmw
 import (
 	"context"
 	"database/sql/driver"
+	"reflect"
 )
 
+//go:generate go run ./tools/rows_picker_gen.go -o rows_picker.go
+
 // Compile time validation that our types implement the expected interfaces
 var (
-	_ driver.Rows                           = wrappedRows{}
-	_ driver.RowsColumnTypeDatabaseTypeName // TODO
-	_ driver.RowsColumnTypeLength           // TODO
-	_ driver.RowsColumnTypeNullable         // TODO
-	_ driver.RowsColumnTypePrecisionScale   // TODO
-	_ driver.RowsColumnTypeScanType         // TODO
-	_ driver.RowsNextResultSet              // TODO
+	_ driver.Rows = wrappedRows{}
 )
 
+// RowsUnwrapper must be used by any middleware that provides its own wrapping
+// for driver.Rows. RowsUnwrap should return the original driver.Rows the
+// middleware received. The result of RowsUnwrap is repeatedly unwrapped
+// until the result no longer implements the interface (this should be
+// the driver.Rows from the original database/sql driver). sqlmw will then
+// pass a type that matches the optional interface set of the driver.Rows
+// implementation of the original driver.
+//
+// If a middleware returns a custom driver.Rows, the custom implmentation
+// must support all the driver.Rows optional interfaces that are supported by
+// by the drivers that will be used with it. To support any arbitrary driver
+// all the optional methods must be supported.
+type RowsUnwrapper interface {
+	RowsUnwrap() driver.Rows
```

Comment:
```suggestion
	Unwrap() driver.Rows
```

Reply:
Since this explicitly unwraps the rows type. and similar techniques could be applied elsewhere (like Tx) , being explicit felt like it made more sense here.

Reply:
@fho's suggestion here feels right to me

re the comment: it could use some work, but it's directionally right.
- let's explain *why* you would wrap a Rows object
- "sqlmw will then pass a type that matches the optional interface set of the driver.Rows implementation of the original driver." this is hard to parse and could use clarity, i don't understand what it's trying to say
- maybe give an example method for the optional methods when it comes to wrapping rows objects